### PR TITLE
Support Vertical Layout Feature

### PR DESCRIPTION
**Pull request by wching**
# #22 

Added the feature to support vertical layout!!
😃 🌮 

#### Things that we can discuss, since I'm not sure:

1. *Naming*? -> Since it not only calculates the x coordinate now. I tried to use just **coordinate** instead of xCoordinate and remove the **X** references on the method names.

2. *Codestyle*? -> This since I formatted and it moved a lot of stuff around 😞  (Tabs vs Spaces) 🚨 😄 

This is how it looks:

### CODE REVIEW

**Comment by rbro112** [comment]
Love it! At work now but I'll take a look into it this afternoon. One thing I might implement here is an option in the demo app to enable vertical scrolling. This will use a vertical recyclerview and viewpager instead of a horizontal one, only showing one indicator at a time. Although this could get tough since this would allow more items to be shown on the screen at the same time, which could affect how the indicator determines what position is currently selected. I'll keep you posted.

**Comment by wching** [comment]
@rbro112 any way I could help you out on this? Want me to try implement the switch on the sample?

**Comment by rbro112** [comment]
@wching Sorry Walter! Been on vacation. I'll make the necessary changes to the demo app. Give me a tiny bit of time and I'll have it done. Likely this weekend. I was going to add an option in the overflow menu in the toolbar to allow for a vertical layout. This will switch the recyclerview/viewpagers to vertical orientations and will likely disable the LTR viewpager. You're welcome to implement that if you'd like!

**Comment by wching** [comment]
no worries @rbro112! Hope you had some great vacations! I haven't had the chance to update the demo app. But I'll try to do it if you haven't done it already :)

**Comment by wching** [comment]
@rbro112 sorry to keep bothering 😅  Do you still want to implement that feature on the sample? wouldn't it be enough to show the vertical layout in the viewpager? 😄

**Comment by rbro112** [comment]
@wching, would love to get the sample. Sorry for the delay, I've been out of the country for quite awhile but I'm back now!

**Comment by wching** [comment]
@rbro112 ok! I'll do it through this week then 👍 thanks for the answer and I hope you had a great time out of the country!

**Comment by wching** [comment]
Updated the sample 💃 

Sorry for the loooooooong time updating this. The 3 fragments reflect the support.

This is how it looks now @rbro112, hope you like it 😄

File: indefinitepagerindicator/src/main/res/values/attrs.xml [hunk 1]
```diff
@@ -10,5 +10,6 @@
         <attr name="selectedDotColor" format="color" />
         <attr name="dotSeparation" format="dimension" />
         <attr name="supportRTL" format="boolean" />
+        <attr name="verticalSupport" format="boolean" />
     </declare-styleable>
 </resources>
```
Comment (wching) [inline indefinitepagerindicator/src/main/res/values/attrs.xml]: New line here

File: sample/src/main/res/layout/fragment_recyclerview_sample.xml [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
+            android:layout_marginStart="8dp"
+            android:visibility="gone"
+            indefinite_pager_indicator:dotColor="@android:color/holo_red_dark"
+            indefinite_pager_indicator:selectedDotColor="@android:color/holo_red_dark"
+            indefinite_pager_indicator:verticalSupport="true"
+            tools:background="@color/cardview_dark_background"/>
+
+    </FrameLayout>
 
 </LinearLayout>
```
Comment (wching) [inline sample/src/main/res/layout/fragment_recyclerview_sample.xml]: New line

File: sample/src/main/res/layout/fragment_recyclerview_sample.xml [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
+    <FrameLayout
         android:layout_width="match_parent"
-        android:layout_height="match_parent" />
+        android:layout_height="wrap_content">
+
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/recyclerview"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_marginEnd="30dp"
```
Comment (wching) [inline sample/src/main/res/layout/fragment_recyclerview_sample.xml]: Maybe use margin 30dp

File: indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt [hunk 4]
```diff
@@ -78,61 +79,85 @@ class IndefinitePagerIndicator @JvmOverloads constructor(context: Context, attrs
...
-    }
+    private fun getDotYCoordinate(): Int = selectedDotRadiusPx
+
+    /**
+     * Calculates the distance between 2 dots center.
+     */
+    private fun getDistanceBetweenTheCenterOfTwoDots() = 2 * dotRadiusPx + dotSeparationDistancePx
 
     /**
      * Calculates a dot's radius based on x position.
```
Comment (wching) [inline indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt]: Update the comment here?

File: indefinitepagerindicator/src/main/res/values/attrs.xml (hunk 1 above)
Comment (wching) [reply indefinitepagerindicator/src/main/res/values/attrs.xml]: 🔒

File: sample/src/main/res/layout/fragment_recyclerview_sample.xml (hunk 2 above)
Comment (wching) [reply sample/src/main/res/layout/fragment_recyclerview_sample.xml]: 🔒

File: sample/src/main/res/layout/fragment_recyclerview_sample.xml (hunk 3 above)
Comment (wching) [reply sample/src/main/res/layout/fragment_recyclerview_sample.xml]: 🔒

File: indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt (hunk 4 above)
Comment (wching) [reply indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt]: 🔒

**Review by wching — COMMENTED** [review]
I reviewed everything again, so I just need another pair of eyes 👀 here in case I'm not seeing something.

**Comment by wching** [comment]
@rbro112 is it ok just to merge? 😅 will love to add other things after this :)

**Comment by rbro112** [comment]
@wching I'm back! Go ahead and merge, I was able to take a look and it looks good to me. I'll be updating more soon as I'm now a full-time Android dev again!

**Review by rbro112 — APPROVED** [review]
👍

**Comment by wching** [comment]
Awesome sauce! Welcome back! 👍

**Comment by rbro112** [comment]
Will be pushing this as v1.0.10 to the maven repo tonight! v1.1 is in the works as well, will add quite a few improvements!

**Comment by rbro112** [comment]
Accidentally pushed version updates to master instead of opening a PR 😬. Not too big of a deal since it's just versioning but bad practice either way!

Nonetheless, the library has been updated to v1.0.10 and now is live on Maven for the world to use. Thanks @wching!!!

**Pull request merged**